distributed/executor.py

```diff
@@ -601,10 +602,23 @@ def _scatter(self, data, workers=None):
 
         raise gen.Return(remotes)
 
+    def _threaded_scatter(self, q_or_i, qout):
+        """ Internal function for scattering Iterable/Queue data
+        """
+        if isinstance(q_or_i, Iterator):
+            get = next
+        if isinstance(q_or_i, pyQueue):
+            get =  pyQueue.get
+
+        while True:
+            d = get(q_or_i)
+            [f] = self.scatter([d])
+            qout.put(f)
+
     def scatter(self, data, workers=None):
         """ Scatter data into distributed memory
 
-        Accepts a list of data elements or dict of key-value pairs
+        Accepts a list/Queue/Iter. of data elements or dict of key-value pairs
```

Comment:
We could use more information about these inputs, and the relationships about the return values.